R/find_anaemia.R

```diff
@@ -0,0 +1,316 @@
+#'    gender, and special condition (like pregnancy). Therefore, one variable
+#'    can indicate the type of sample population characteristic requirement for
+#'    anaemia diagnosis. The required population group for this variable is
+#'    mentioned in the below table.
+#'
+#'    | **Population** | **Valye** |
+#'    | :--- | :--- |
+#'    | Children 6-59 months of age |	1 |
+#'    | Children 5-11 years of age |	2 |
+#'    | Children 12-14 years of age	| 3 |
+#'    | Non-pregnant women (15 years and above) | 4 |
+#'    | Pregnant women	| 5 |
+#'    | Men (15 years and above) |	6 |
+#'
+#'    if the dataset did not include this variable yet, please create one before
+#'    using this function. Sample r code for generating this variable is
+#'    mentioned below using the sample dataset `hbData`.
+#'
+#' df$pop_group <- with(df,ifelse(
+#'  m.age >= 6 & m.age < 60, 1, ifelse(
+#'  ch.age >= 5 & ch.age < 12, 2, ifelse(
+#'  ch.age >= 12 & ch.age < 15, 3, ifelse(
+#'  ch.age >= 15 & !is.na(ch.age) & sex == 2, 4, ifelse(
+#'  group == "Pregnant Not Principal Carer" |
+#'    group == "Pregnant and lactating Principal Carer" |
+#'    group == "Pregnant Principal carer", 5, ifelse(
+#'  ch.age >= 15 & !is.na(ch.age) & sex == 1, 6, NA)))))))
+#'
+#'
+#' @param hb Sample observation's haemoglobin level from the dataset. The Hb
+#'    values should record in the grams per liter (g/L) units.
+#' @param add This parameter's default option is TRUE and will add new
+#'    generated variables `anaemia_all` to your existing dataset applied in
+#'    this function. This newly developed categorical variable comprises three
+#'    types of flag categories resulting from data checking results; no anaemia,
+#'     mild anaemia, moderate anaemia, and severe anaemia. The following table
+#'     explains the cut-off points applied in this diagnostic function.
+#'
+#'    | **Population** | **Mild** |	**Moderate** | **Severe** |
+#'    | :--- | :--- | :--- | :--- |
+#'    | Children 6-59 months of age |	100 - 109 |	70 - 99 |	< 70 |
+#'    | Children 5-11 years of age |	110 - 114	| 80 - 109 |	< 80 |
+#'    | Children 12-14 years of age	| 110 - 119	| 80 - 109 |	< 80 |
+#'    | Non-pregnant women |  |  |  |
+#'    | (15 years and above) | 110 - 119	| 80 - 109 | < 80 |
+#'    |Pregnant women	| 100 - 109	| 70 - 99	| < 70 |
+#'    |Men |  |  |  |
+#'    |(15 years and above) |	110 - 129 |	80 - 109 |	< 80 |
+#'
+#'
+#' @return anaemia   A data frame with the same structure as `df` is named
+#'    `anaemia`. In this new data.frame, the new variable `anaemia_all` can be
+#'    observed, containing the information which observation was affected by
+#'    what type of anaemia: mild, moderate, or severe.
+#'
+#' @examples
+#'   flag_anaemia(df = df,
+#'                pop_group = pop_group,
+#'                hb = hb
+#'                add = TRUE)
+#'
+#' @export
+#'
+#'
+#
+################################################################################
+
+cut()
```

Comment:
Good work on this. This is already looking much better than the previous one. Well done!

As discussed, look into refactoring this function using the following logicL

```
################################################################################
# group specific functions
#' 
#' @param hb A numeric value or a vector of numeric values of Haemoglobin (Hb)
#'   in g/L for pregnant women.


find_anaemia_preg <- function(hb) {
  ## Use cut function with breaks for pregnant women
}








################################################################################
## Overall function

find_anaemia <- function(hb, group = c("group1", "group2", "group3")) {
  group <- match.arg(group)
  
  if (group == "group1") {
    find_anaemia_group1(hb = hb)
  }
  
  if (group == "group2") {
    find_anaemia_group2(hb = hb)
  } 
}
```